blocks/identity-block/features/login/default.jsx

```diff
@@ -29,11 +29,21 @@ export function definedMessageByCode(code) {
 }
 
 const Login = ({ customFields }) => {
-	const { redirectURL, redirectToPreviousPage, loggedInPageLocation, OIDC, termsAndPrivacyURL } =
-		customFields;
+	const {
+		redirectURL,
+		redirectToPreviousPage,
+		loggedInPageLocation,
+		OIDC,
+		termsAndPrivacyURL,
+	} = customFields;
 
-	const urlString = window.location.href;
-	const url = new URL(urlString);
+	let urlString = '';
+	let url = '';
+
+	if (window?.location?.href) {
+		urlString = window.location.href;
+		url = new URL(urlString);
+	}
 
 	const { isAdmin, arcSite } = useFusionContext();
```

Comment:
@accbjt can you make sure that `isAdmin` is `true` here when you are on the preview page in your test environment? This is how Fusion engine identifies when it's being rendered in the the PB Editor, and it should always be true in that case. If it's not then there's a bug somewhere in Engine.

Reply:
And to be clear, when you say "preview" do you mean viewing it the PB Editor?

Reply:
No not on PB editor directly but when you click on the eye icon to preview the page. Here is the below link to the preview page.

http://localhost/pf/login/?_website=the-gazette&outputType=default

Reply:
Okay, I mis-understood the point of that logic then. It'll need to be there then, there's not really any way around checking the referrer URL in that specific case. I'll approve this and add notes for the CodeQL alerts that they'll be addressed in that new ticket you made.